pretrain.py

```diff
@@ -21,23 +22,41 @@
 dataset = load_dataset(config.data)
 
 
-### Train the tokenizer & Load the pretrained tokenizer
+### Train the tokenizer & Load the pretrained tokenizer ###
 # ! Multiple Possible Tokenizers
 # TODO: Find the best tokenizer
 
 # You can choose the tokenizer type, default is bpe
 tokenizer = train_tokenizer(dataset=dataset, tokenization_type=config.tokenizer, vocab_size=config.vocab_size)
 
-### TODO: Tokenize the dataset
+### Setup Model ###
 
-### TODO: Init the config and the model
-
-config = ...
-model = ...
-data_collator = ...
+esm_config = EsmConfig(
+    vocab_size=config.vocab_size,
+    num_hidden_layers=config.n_layers,
+    hidden_size=config.n_dims,
+    num_attention_heads=config.n_heads,
+)
+model = TFEsmForMaskedLM(
+    config=esm_config,
+)
+data_collator = DataCollatorForLanguageModeling(tokenizer=tokenizer, mlm=True, mlm_probability=0.15)
 
 ### Setup trainer ###
 
-training_args = ...
-trainer = ...
+training_args = TrainingArguments(
+    output_dir="/scratch/output",
+    num_train_epochs=config.epochs,
+    per_device_train_batch_size=config.batch_size,
+)
+
+#! Using swissprot (Small DB)
+trainer = Trainer(
+    model=model,
+    args=training_args,
+    data_collator=data_collator,
+    tokenizer=tokenizer,
+    training_dataset=dataset["train"]["sequence"],
```

Comment:
Why the `["sequence"]` key? Could you describe the intended structure of `dataset["train"]` and `dataset["validation"]`?

Reply:
The dataset is composed of multiple files: `test.parquet`, `train.parquet`, and `valid.parquet` each composed of different sequences. The function/library I'm using (`datasets.load_dataset`) parses it into different attributes in a class. You can get them by fetching train, test, or valid.

In addition, each file is composed of different columns, the one we need being sequence which is why we narrow down our search to it.

Reply:
Put it into the code and it's `["Sequence"]`